### ISSUE TITLE
Omit aliases in joins?

### ISSUE THREAD

**Issue by hadley** [tidyverse/dbplyr#960]
It just occurred to me that in this query:

``` r
library(dbplyr)
library(dplyr, warn.conflicts = FALSE)

lf1 <- lazy_frame(a = 1, b = 2, c = 1, .name = "lf1")
lf2 <- lazy_frame(a = 1, b = 2, z = 1, .name = "lf2")

lf1 |> left_join(lf2)
#> Joining, by = c("a", "b")
#> <SQL>
#> SELECT `LHS`.*, `z`
#> FROM `lf1` AS `LHS`
#> LEFT JOIN `lf2` AS `RHS`
#>   ON (`LHS`.`a` = `RHS`.`a` AND `LHS`.`b` = `RHS`.`b`)
```

<sup>Created on 2022-08-04 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

Since the LHS and the RHS of the join are simple tables, we could drop the aliases, and write:

```R
#> <SQL>
#> SELECT `lf1`.*, `z`
#> FROM `lf1`
#> LEFT JOIN `lf2`
#>   ON (`lf1`.`a` = `lf2`.`a` AND `lf1`.`b` = `lf2`.`b`)
```

**Comment by mgirlich** [comment]
I think this will be solved via #892 😉

**Comment by hadley** [comment]
Ok, I knew I had seen it somewhere!
